plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/storage/TestTrinoHudiStorage.java

```diff
@@ -293,6 +294,65 @@ private static void validatePathInfoList(
         }
     }
 
+    @Test
+    void testListEmptyDirectoryOnHierarchicalFileSystem()
+            throws IOException
+    {
+        HoodieStorage storage = getStorage();
+
+        // Create an empty directory on hierarchical filesystem (LocalFileSystem)
+        StoragePath emptyDir = new StoragePath(getTempDir(), "empty_directory");
+        assertThat(storage.createDirectory(emptyDir)).isTrue();
+        assertThat(storage.exists(emptyDir)).isTrue();
+
+        // On hierarchical filesystem, listing empty directory should return empty list
+        List<StoragePathInfo> entries = storage.listDirectEntries(emptyDir);
+        assertThat(entries).isEmpty();
+
+        List<StoragePathInfo> files = storage.listFiles(emptyDir);
+        assertThat(files).isEmpty();
+    }
+
+    @Test
+    void testListEmptyDirectoryOnNonHierarchicalFileSystem()
+            throws IOException
```

Comment:
Remove redundant `throws IOException` (we need this line if we use try-with-resources, though)

Reply:
fixed